CSharp/DevVmPowershell/Helpers/Implementations/ProcessingHelper.cs

```diff
@@ -353,45 +235,22 @@ public async Task<bool> DeleteWorkerManagerServerAsync()
 
-			// Setup for checking if Worker Manager Server exists
-			Relativity.Services.TextCondition conditionServer = new Relativity.Services.TextCondition()
-			{
-				Field = Constants.Processing.NameField,
-				Operator = Relativity.Services.TextConditionEnum.EqualTo,
-				Value = Constants.Processing.ResourceServerName
-			};
-
-			Relativity.Services.Query queryServer = new Relativity.Services.Query()
-			{
-				Condition = conditionServer.ToQueryString()
-			};
-
-			using (IWorkerServerManager workerServerManager = ServiceFactory.CreateProxy<IWorkerServerManager>())
+			HttpClient httpClient = RestHelper.GetHttpClient(ConnectionHelper.RelativityInstanceName, ConnectionHelper.RelativityAdminUserName, ConnectionHelper.RelativityAdminPassword);
+			// Check if Worker Manager Server is already exists
+			int workerManagerServerId = await GetWorkerManagerServerArtifactIdAsync(httpClient);
+			if (workerManagerServerId != -1)
+			{
+				// Delete Worker Manager Server
+				await CallDeleteWorkerManagerServerAsync(httpClient, workerManagerServerId);
+				Console.WriteLine($"{nameof(DeleteWorkerManagerServerAsync)} - Successfully deleted Worker Manager Server ({Constants.Processing.ResourceServerName})");
+				wasDeleted = true;
+			}
+			else
 			{
-				WorkerServerQueryResultSet queryResult = await workerServerManager.QueryAsync(queryServer);
-
-				if (queryResult.Success && queryResult.Results.Exists(x => x.Artifact.ServerType.Name.Equals(Constants.Processing.WorkerManagerServer, StringComparison.OrdinalIgnoreCase)))
-				{
-					await workerServerManager.DeleteSingleAsync(queryResult.Results.Find(x => x.Artifact.ServerType.Name.Equals(Constants.Processing.WorkerManagerServer, StringComparison.OrdinalIgnoreCase)).Artifact.ArtifactID);
-
-					queryResult = await workerServerManager.QueryAsync(queryServer);
-
-					if (queryResult.Success && !queryResult.Results.Exists(x => x.Artifact.ServerType.Name.Equals(Constants.Processing.WorkerManagerServer, StringComparison.OrdinalIgnoreCase)))
-					{
-						wasDeleted = true;
-						Console.WriteLine($"{nameof(DeleteWorkerManagerServerAsync)} - Successfully deleted Worker Manager Server ({Constants.Processing.ResourceServerName})");
-					}
-				}
-				else if (queryResult.Success && queryResult.TotalCount == 0)
-				{
-					Console.WriteLine($"{nameof(DeleteWorkerManagerServerAsync)} - Failed to delete Worker Manager Server ({Constants.Processing.ResourceServerName}) as it doesn't exist");
-					wasDeleted = true;
-				}
-				else
-				{
-					Console.WriteLine($"{nameof(DeleteWorkerManagerServerAsync)} - Failed to delete and check for Worker Manager Server ({Constants.Processing.ResourceServerName})");
-				}
+				Console.WriteLine($"{nameof(DeleteWorkerManagerServerAsync)} - Failed to delete Worker Manager Server ({Constants.Processing.ResourceServerName}) as it doesn't exist");
+				wasDeleted = true;

```

Comment:
Can we set this variable (wasDeleted) to true only once after the if/else statement, instead of doing it individually in each statement just like we did with the wasChoiceAddedToPool variable above?

Reply:
Done